src/ConfigAggregator.php

```diff
@@ -231,4 +237,41 @@ private function cacheConfig(array $config, $cachedConfigFile)
             var_export($config, true)
         ));
     }
+
+    /**
+     * @param array $config
+     *
+     * @return array
+     */
+    private function postProcessConfig(array $config)
+    {
+        foreach ($this->postProcessors as $processor) {
+            $config = $processor($config);
+        }
+
+        return $config;
+    }
+
+    /**
+     * @param Closure|object|callable $provider
+     *
+     * @return string
+     */
+    private function detectProviderType($provider)
+    {
+        if ($provider instanceof Closure) {
+            return 'Closure';
+        }
+
+        if (is_callable($provider)) {
+            return is_string($provider) ? $provider : gettype($provider);
+        }
+
+        if (is_object($provider)) {
```

Comment:
By the way, I would prefer doing this check before we check if the provider is callable. If the provider is a callable object, the information `object` is not really informational as we could provide the class name instead.

I just applied the same logic I've found within the `mergeConfig` method but I think switching the detection is more helpful as I think the class name is more informational as `object`.